public/pages/VisualCreatePolicy/components/ISMTemplate/ISMTemplate.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { ChangeEvent, useState } from "react";
+import { EuiButton, EuiFormRow, EuiComboBox, EuiFlexGroup, EuiFlexItem, EuiFieldNumber } from "@elastic/eui";
+import "brace/theme/github";
+import "brace/mode/json";
+import { ISMTemplate as ISMTemplateData } from "../../../../../models/interfaces";
+import { ISM_TEMPLATE_INPUT_MAX_WIDTH } from "../../utils/constants";
+
+interface ISMTemplateProps {
+  template: ISMTemplateData;
+  onUpdateTemplate: (template: ISMTemplateData) => void;
+  onRemoveTemplate: () => void;
+  isFirst: boolean;
+}
+
+const ISMTemplate = ({ template, onUpdateTemplate, onRemoveTemplate, isFirst }: ISMTemplateProps) => {
+  // TODO: Move this top top of form submition
+  const [isInvalid, setInvalid] = useState(false);
+  return (
+    <EuiFlexGroup gutterSize="l" alignItems="center">
+      <EuiFlexItem style={{ maxWidth: ISM_TEMPLATE_INPUT_MAX_WIDTH }}>
+        <EuiFormRow isInvalid={false} error={null}>
+          <EuiComboBox
+            placeholder="Add index patterns"
+            noSuggestions
+            selectedOptions={template.index_patterns.map((pattern) => ({ label: pattern }))}
+            onChange={(selectedOptions) => {
+              onUpdateTemplate({ ...template, index_patterns: selectedOptions.map(({ label }) => label) });
+              setInvalid(false);
+            }}
+            onCreateOption={(searchValue) => {
+              if (!searchValue.trim()) {
+                return false;
+              }
+
+              if (searchValue.includes(" ")) {
+                setInvalid(false);
+                return;
+              }
```

Comment:
Confused here, why this `setInvalid(false)`, but on line 61 is `setInvalid(true)`

And do we need `onSearchChange` here, seems we dont need to do search for ISM template

Reply:
https://user-images.githubusercontent.com/46505179/129070382-5483135e-1c55-44ad-9d38-ce3f117f1f65.mov

If the value has a space in it then we don't actually create the option -> the input is cleared -> the invalid status is set back to false because there is no more input anymore.

This will all be reworked in later PRs to integrate it w/ create button though also.